nsxt/resource_nsxt_policy_context_profile.go

```diff
@@ -0,0 +1,486 @@
+		var err error
+		client := infra.NewDefaultContextProfilesClient(connector)
+		obj, err = client.Get(id)
+		if err != nil {
+			return handleReadError(d, "ContextProfile", id, err)
+		}
+	}
+
+	d.Set("display_name", obj.DisplayName)
+	d.Set("description", obj.Description)
+	setPolicyTagsInSchema(d, obj.Tags)
+	d.Set("nsx_id", id)
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+	d.Set("attribute", fillAttributesInSchema(obj.Attributes))
+	myset := d.Get("attribute").(*schema.Set)
+	print("attributestrings!!!!\n")
+	print(myset.GoString())
+	print("\n")
+	for _, attr := range myset.List() {
+		attributeMap := attr.(map[string]interface{})
+		subAttributes := attributeMap["sub_attribute"].(*schema.Set)
+		print("subattributestrings!!!!\n")
+		print(subAttributes.GoString())
+		print("\n")
+	}
+	return nil
+}
+
+func resourceNsxtPolicyContextProfileUpdate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining ContextProfile ID")
+	}
+
+	// Read the rest of the configured parameters
+	description := d.Get("description").(string)
+	displayName := d.Get("display_name").(string)
+	tags := getPolicyTagsFromSchema(d)
+	attributes := d.Get("attribute").(*schema.Set).List()
+	err := checkAttributesValid(attributes, m)
+	if err != nil {
+		return err
+	}
+	var attributesStructList []model.PolicyAttributes
+	attributesStructList, err = constructAttributesModelList(attributes)
+	if err != nil {
+		return err
+	}
+
+	obj := model.PolicyContextProfile{
+		DisplayName: &displayName,
+		Description: &description,
+		Tags:        tags,
+		Attributes:  attributesStructList,
+	}
+
+	// Update the resource using PATCH
+	if isPolicyGlobalManager(m) {
+		rawObj, err1 := convertModelBindingType(obj, model.PolicyContextProfileBindingType(), gm_model.PolicyContextProfileBindingType())
+		if err1 != nil {
+			return err1
+		}
+		gmObj := rawObj.(gm_model.PolicyContextProfile)
+		client := gm_infra.NewDefaultContextProfilesClient(connector)
+		err = client.Patch(id, gmObj)
+	} else {
+		client := infra.NewDefaultContextProfilesClient(connector)
+		err = client.Patch(id, obj)
+	}
+
+	if err != nil {
+		return handleUpdateError("ContextProfile", id, err)
+	}
+
+	return resourceNsxtPolicyContextProfileRead(d, m)
+}
+
+func resourceNsxtPolicyContextProfileDelete(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining ContextProfile ID")
+	}
+
+	connector := getPolicyConnector(m)
+	var err error
+	force := true
+	if isPolicyGlobalManager(m) {
+		client := gm_infra.NewDefaultContextProfilesClient(connector)
+		err = client.Delete(id, &force)
+	} else {
+		client := infra.NewDefaultContextProfilesClient(connector)
+		err = client.Delete(id, &force)
+	}
+	if err != nil {
+		return handleDeleteError("ContextProfile", id, err)
+	}
+
+	return nil
+}
+
+func getContextProfilePolicyAttributesSchema() *schema.Schema {
```

Comment:
I think this should be at the top of the file. just before or after resourceNsxtPolicyContextProfile. This is the way it is done with most of the resources and it makes it easier to understand the structure

Reply:
Sure. Thanks.